scripts/fluentd-queue-report

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+Usage: [PER_POD=true] $0
+Using PER_POD=true will list the fluentd stats for each fluentd pod
+
+This command will look at the fluentd buffer queue files on each fluentd
+pod, sum the sizes of those files, find the oldest and newest, in order
+to find the per-pod and total aggregate fluentd queue size, which is
+the amount of data read by fluentd and pending to be sent to 
+Elasticsearch.
+
+green - oldest file is less than 1 minute old
+yellow - oldest file is less than 5 minutes old
+red - oldest file is more than 5 minutes old
+These numbers are rather arbitrary, so may require some adjustment
+for your environment.
+
+If you see a lot of pods in the red state, or you see the average
+and total size increasing, use PER_POD=true to see if there are
+some problematic pods.  You may need to investigate:
+
+    oc exec fluentd-xxx -- logs
+    oc exec fluentd-xxx -- ls -lrtFi /var/lib/fluentd
+    oc exec fluentd-xxx -- ls -lrtFi /var/lib/fluentd/es-retry
+    oc exec fluentd-xxx -- ls -lrtFi /var/lib/fluentd/buffer-output-es-config
+
+and possibly restart those fluentd pods (e.g. oc delete pod fluentd-xxx)
+
+output looks something like this with PER_POD=true - without, only SUMMARY
+will be printed:
+
+STATUS OLDEST NEWEST       SIZE POD                   NODETYPE NODE
+green      20      1          0 logging-fluentd-xxxxx compute  ip-xxx.compute.internal
+yellow    100      5       5432 logging-fluentd-yyyyy infra    ip-yyy.compute.internal
+red      1234      9   11115920 logging-fluentd-zzzzz master   ip-zzz.compute.internal
+...
+
+SUMMARY
+
+TIME                       PODS RED YELLOW GREEN OLDEST TOTAL_SIZE    LARGEST    AVERAGE
+2019-09-13T02:35:42+0000     33   4      9    20   3457  333748745    9833954   10113598
+
+OLDEST  buffer file was on pod logging-fluentd-zzzzz node master  ip-zzz.compute.internal
+LARGEST buffer file was on pod logging-fluentd-qqqqq node compute ip-qqq.compute.internal
+
+EOF
+}
+
+if [ -n "${1:-}" ] ; then
+    usage
+    exit 0
+fi
+
+# argument is how many seconds old the oldest
+# buffer file is
+color() {
```

Comment:
Can this maybe be extracted to be used elsewhere?  not exactly sure which other scripts it where it could be useful

Reply:
not really - this is very specific to the usage in this file